src/Compositor.hpp

```diff
@@ -32,6 +32,7 @@
 #include <aquamarine/backend/Backend.hpp>
 #include <aquamarine/output/Output.hpp>
 
+struct SImageDescription;
```

Comment:
I don't think this forward def is legal when you use it as a return type. Will fail no-pch